### PR TITLE
"add: add authorisation flow"

### PR DESCRIPTION
**Pull request by alonakaliuzhna**
### Type of change:

- [x] Feature (introduces new functionality)
- [ ] Bugfix
- [x] Configuration
- [ ] Refactoring (doesn't change functionality, only the code)
- [ ] Style changes (doesn't affect functionality, only the look of the components)

### PR Checklist:

- [ ] Synced up with latest from the main branch/Code is linted
- [x] Description of the changes has been added (Screenshots/Video)
- [ ] Chakra components/constants/hooks have been used wherever possible
> If not explain here
- [ ] Commit messages follow [conventional commits guideline](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#summary)
- [x] Console has been checked for warnings

#### Screenshots/Video before change
Add screenshots from the [current deployed state](https://sandbox-money-app.herokuapp.com/) of the changed components here.

#### Screenshots/Video after change
Add screenshots with the changes applied here. 

https://user-images.githubusercontent.com/83176493/181260302-7550a448-2dbf-4d36-9f7a-0b7ad9d8abf0.mp4

before authorisation
cd37b44579c522ebec0e82f423a08376

after 
58943d350c82b5d85a78d3004841f734

### CODE REVIEW

File: src/components/LoginForm/LoginForm.jsx [hunk 1]
```diff
@@ -20,13 +22,23 @@ import i18next from 'i18next';
 
 import { LogoWalletIcon } from '@/assets';
 
+import loginAction from '../../api/AuthProvider';
```
Comment (itsnotpolza) [inline src/components/LoginForm/LoginForm.jsx:25]:
```suggestion
import loginAction from '@/api/AuthProvider';
```

File: src/pages/App.jsx [hunk 2]
```diff
@@ -9,19 +10,23 @@ import { Fallback } from './Fallback';
...
-          <Route path="*" element={<Fallback />} />
-          <Route path="/expenses" element={<Expenses />} />
-        </Route>
-      </Routes>
-    </BrowserRouter>
+    <QueryClientProvider client={queryClient}>
+      <BrowserRouter>
+        <Routes>
+          <Route path="/login" element={<Login />} index />
+          <Route element={<AppLayout />}>
```
Comment (itsnotpolza) [inline src/pages/App.jsx:21]: close all routes except login until user is authenticated

File: src/api/AuthHook.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { useEffect, useState } from 'react';
+
+import loginAction from './AuthProvider';
+export default function useLoginAction() {
```
Comment (lazzzaroni) [inline src/api/AuthHook.js:4]: Add empty space between import and export. It's for the better readability :)

File: src/components/Header/Header.jsx [hunk 4]
```diff
@@ -22,6 +24,8 @@ export const Header = () => {
   const headerTextColor = useColorModeValue('teal.900', 'orange.300');
   const iconsThemeColor = useColorModeValue('teal.900', 'orange.300');
   const iconsHoverThemeColor = useColorModeValue('teal.900', 'orange.300');
+
+  const user = useQuery(['user'], () => loginAction());
   return (
```
Comment (lazzzaroni) [inline src/components/Header/Header.jsx:29]: Also here I would add empty space

**Review by kalashnikau — APPROVED** [review]

**Review by lazzzaroni — DISMISSED** [review]

**Review by itsnotpolza — APPROVED** [review]

**Review by lazzzaroni — APPROVED** [review]

**Pull request merged**